redfish-core/lib/assembly.hpp

```diff
@@ -247,6 +248,34 @@ inline void getReadyToRemoveOfTodBattery(
                         assemblyIndex));
 }
 
+void getAssemblyPresence(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                         const auto& serviceName, const auto& assembly,
+                         const auto& assemblyIndex)
+{
+    nlohmann::json& assemblyArray = asyncResp->res.jsonValue["Assemblies"];
+    nlohmann::json& assemblyData = assemblyArray.at(assemblyIndex);
+
+    assemblyData["Status"]["State"] = "Enabled";
+
+    sdbusplus::asio::getProperty<bool>(
+        *crow::connections::systemBus, serviceName, assembly,
+        "xyz.openbmc_project.Inventory.Item", "Present",
+        [asyncResp, assemblyIndex,
+         &assemblyData](const boost::system::error_code ec, const bool value) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR("DBUS response error: {}", ec.value());
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
```

Comment:
Add json reference here using 'assemblyIndex'
```
  nlohmann::json& assemblyData = assemblyArray.at(assemblyIndex);
```

Reply:
I think 'assemblyData` should not be captured for the async-method-call, as json reference may be changed if there is another async call method which modifies json structure.
So, capture only 'assemblyIndex', and use it to get assemblyData inside the callback.
If necessary, use the different variable name (for assemblyData).

Reply:
done